pjsip/src/pjsua-lib/pjsua_call.c

```diff
@@ -1453,6 +1453,155 @@ on_incoming_call_med_tp_complete(pjsua_call_id call_id,
     return on_incoming_call_med_tp_complete2(call_id, info, NULL, NULL, NULL);
 }
 
+static void rejected_incoming_call_cb(pjsip_rx_data *rdata, pjsip_tx_data *tdata, 
+                                      int st_code)
+{
+    /* Call on_rejected_incoming_call() only if to-tag is not set. */
+    if ((rdata->msg_info.to->tag.slen == 0) &&
+        pjsua_var.ua_cfg.cb.on_rejected_incoming_call)
+    {
+        pjsip_from_hdr* from_hdr;
+        pjsip_to_hdr* to_hdr;
+        pjsip_sip_uri* uri;
+        pjsua_on_rejected_incoming_call_param param;
+
+        pj_bzero(&param, sizeof(pjsua_on_rejected_incoming_call_param));
+        if (st_code == 0)
+            st_code = tdata->msg->line.status.code;
+
+        param.rdata = rdata;
+        param.tdata = tdata;
+        param.code = st_code;
+
+        from_hdr = rdata->msg_info.from;
+        if (from_hdr) {
+            param.remote_info.ptr = param.buf_.remote_info;
+            uri = (pjsip_sip_uri*)pjsip_uri_get_uri(from_hdr->uri);
+            param.remote_info.slen = pjsip_uri_print(PJSIP_URI_IN_FROMTO_HDR,
+                uri,
+                param.buf_.remote_info,
+                sizeof(param.buf_.remote_info));
+        }
+
+        to_hdr = rdata->msg_info.to;
+        if (to_hdr) {
+            param.local_info.ptr = param.buf_.local_info;
```

Comment:
As far as I understand, `buf_` is only used locally, and has to live for the lifetime of the callback below.
So why not having it locally only (nicer `pjsua_on_rejected_incoming_call_param`)? Why not using a memory pool?

Reply:
It is using the same pattern as in `pjsua_call_info` (using `buf_` as internal buffer).

Reply:
I see. I am curious why. :-)